### ISSUE TITLE
Unable to read SRC7 metadata via the TS SDK

### ISSUE THREAD

**Issue by Dhaiwat10** [FuelLabs/fuels-ts#1880]
> Incoming from this forum post: https://forum.fuel.network/t/metadat-read-fail/4197

When we try to read `Metadata` from a SRC7-compliant contract, the SDK crashes with the following error:

```
Uncaught (in promise) RevertError: The script reverted with reason Unknown
    at revertErrorFactory (index.mjs:206:14)
    at RevertErrorCodes.getError (index.mjs:235:12)
    at RevertErrorCodes.assert (index.mjs:219:30)
    at new ScriptResultDecoderError (index.mjs:265:43)
    at decodeCallResult (index.mjs:375:11)
    at decodeContractCallScriptResult (index.mjs:576:95)
    at FunctionInvocationResult.getDecodedValue (index.mjs:703:28)
    at new InvocationResult (index.mjs:677:23)
    at new FunctionInvocationResult (index.mjs:746:5)
    at FunctionInvocationResult.build (index.mjs:764:22)
    at async Object.readMetadata [as onPress] (index.tsx:47:17)
```

Setting `Metadata` works fine though.

**Comment by Torres-ssf** [comment]
@Dhaiwat10 I was investigating this yesterday and I am going to share my findings with you:

Just like we already validated, setting the `Metadata` works, but when trying to get it, the TX is reverted.

The `Metadata` enum is defined [here](https://github.com/FuelLabs/sway-standards/blob/b6f76e15b353cb59512072fadac027ead32956b8/standards/src7-metadata/src/src7.sw#L35-L44)

So when setting a `Metadata` we have the option to do so using one of these types: `Int`, `String`, `B256`, and `Bytes`.

On the forum post, the error is happening when using the `String` type. So I've tried the other 3:

1 `Int`: Works both when setting and retrieving it, e.g:
```ts
const metadata = {
  Int: 100,
}
```


2 `B256`: Also works both when setting and retrieving it. e.g:
```ts
const metadata = {
  B256: getRandomB256(),
}
```

3 `Bytes`: Works when setting and retrieving it, however, the returned result is always an empty array.

```ts
const metadata = {
  Bytes: [40, 41, 42],
}
```
It is weird that the returned value by the fn `metadata` is an empty array. It is worth validating if the issue is on the TS-SDK side.
> cc @danielbate 

4 `String`: Works for setting but it always reverts when retrieving. I have also tried some different strings.
```ts
const metadata = {
  String: "fuel",
}
```
I am inclined to believe that the problem here is not coming from the TS-SDK side. 

I hope these findings might help you in some way 🫡

**Comment by danielbate** [comment]
The problem here is that `bytes` in this context is a nested heap type, not supported in `v0`. This is solved in `v1` and proved in #1832.

**Comment by chappie1998** [comment]
@danielbate, when will there be a release with this fixed?

**Comment by danielbate** [comment]
@chappie1998 I'm afraid we can't give an exact release date for this at the moment. The new encoding scheme is part of a wider piece of work, so requires aligning changes and releases with many products in our eco-system.

But as you can see, it is [in active development](https://github.com/FuelLabs/fuels-ts/pull/1832) and we will keep this issue up to date and can coordinate with yourself. #1672 also details the `v1` changes within the TS SDK realm.

**Comment by chappie1998** [comment]
Hey, @danielbate, are these changes out with the latest release on both the sway and SDK side?

**Comment by danielbate** [comment]
So this was included in `v0.80.0` of the [SDK](https://github.com/FuelLabs/fuels-ts/releases/tag/v0.80.0). And `v0.53.0` of `forc`. You will need to build your programs with `--experimental-new-encoding`. 

As the flag suggests, this work is currently experimental so you may experience some unexpected behaviour, but it would be really helpful for you to try it out and feedback any problems.

One issue we are already aware of tho, is that it currently does not support `multicall`, that will be included in #2050.
